etna/transforms/decomposition/base_change_points.py

```diff
@@ -0,0 +1,94 @@
+from abc import ABC
+from abc import abstractmethod
+from typing import List
+from typing import Tuple
+from typing import Type
+
+import pandas as pd
+from ruptures.base import BaseEstimator
+from ruptures.costs import CostLinear
+from sklearn.base import RegressorMixin
+
+TTimestampInterval = Tuple[pd.Timestamp, pd.Timestamp]
+TDetrendModel = Type[RegressorMixin]
+
+
+class BaseChangePointsModelAdapter(ABC):
+    """BaseChangePointsModelAdapter is the base class for transforms with change points."""
+
+    def __init__(self, change_point_model: BaseEstimator, **change_point_model_predict_params):
+        """Init BaseChangePointsModelAdapter.
+
+        Parameters
+        ----------
+        change_point_model:
+            model to get change points
+        change_point_model_predict_params:
+            params for ``change_point_model.predict`` method
+        """
+        self.change_point_model = change_point_model
+        self.change_point_model_predict_params = change_point_model_predict_params
+
+    @abstractmethod
+    def get_change_points(self, df: pd.DataFrame, in_column: str) -> List[pd.Timestamp]:
+        """Find change points in given dataframe and column."""
+        pass
+
+    @staticmethod
+    def _build_intervals(change_points: List[pd.Timestamp]) -> List[TTimestampInterval]:
+        """Create list of stable intervals from list of change points."""
+        change_points = sorted(change_points)
```

Comment:
I left the [comment](https://github.com/tinkoff-ai/etna/pull/822#discussion_r943131191) about changing the implementation